py-polars/polars/datatypes/_parse.py

```diff
@@ -78,7 +78,8 @@ def parse_py_type_into_dtype(input: PythonDataType | type[object]) -> PolarsData
         return Boolean()
     elif input is date:
         return Date()
-    elif input is datetime:
+    elif isinstance(input, type) and issubclass(input, datetime):  # type: ignore[redundant-expr]
+        # pd.Timestamp subclasses `datetime.datetime`
```

Comment:
mypy says the first check is redundant, although removing it makes some tests fail with
```
TypeError: issubclass() arg 1 must be a class
```